evm/test/PerChainTransceivers.t.sol

```diff
@@ -0,0 +1,789 @@
+
+        // Register peer contracts for the nttManager and transceiver. Transceivers and nttManager each have the concept of peers here.
+        nttManagerChain1.setPeer(
+            chainId3, bytes32(uint256(uint160(address(nttManagerChain3)))), 9, type(uint64).max
+        );
+        nttManagerChain3.setPeer(
+            chainId1, bytes32(uint256(uint160(address(nttManagerChain1)))), 7, type(uint64).max
+        );
+
+        // Create the second transceiver, from chain 3 to chain 1.
+        WormholeTransceiver secondWormholeTransceiverChain3Implementation = new MockWormholeTransceiverContract(
+            address(nttManagerChain3),
+            address(wormhole),
+            address(relayer),
+            address(0x0),
+            FAST_CONSISTENCY_LEVEL,
+            GAS_LIMIT
+        );
+        secondWormholeTransceiverChain3 = MockWormholeTransceiverContract(
+            address(new ERC1967Proxy(address(secondWormholeTransceiverChain3Implementation), ""))
+        );
+
+        // Actually initialize properly now
+        secondWormholeTransceiverChain3.initialize();
+
+        nttManagerChain3.setTransceiver(address(secondWormholeTransceiverChain3));
+
+        // Set peers for the transceivers
+        wormholeTransceiverChain1.setWormholePeer(
+            chainId3, bytes32(uint256(uint160(address(wormholeTransceiverChain3))))
+        );
+        wormholeTransceiverChain3.setWormholePeer(
+            chainId1, bytes32(uint256(uint160(address(wormholeTransceiverChain1))))
+        );
+        wormholeTransceiverChain3.setWormholePeer(
+            chainId2, bytes32(uint256(uint160(address(wormholeTransceiverChain2))))
+        );
+        wormholeTransceiverChain2.setWormholePeer(
+            chainId3, bytes32(uint256(uint160(address(wormholeTransceiverChain3))))
+        );
+        secondWormholeTransceiverChain1.setWormholePeer(
+            chainId3, bytes32(uint256(uint160(address(secondWormholeTransceiverChain3))))
+        );
+        secondWormholeTransceiverChain3.setWormholePeer(
+            chainId1, bytes32(uint256(uint160(address(secondWormholeTransceiverChain1))))
+        );
+        secondWormholeTransceiverChain2.setWormholePeer(
+            chainId3, bytes32(uint256(uint160(address(secondWormholeTransceiverChain3))))
+        );
+        secondWormholeTransceiverChain3.setWormholePeer(
+            chainId2, bytes32(uint256(uint160(address(secondWormholeTransceiverChain2))))
+        );
+
+        // Set the thresholds.
+        require(nttManagerChain1.getThreshold() != 0, "Threshold is zero with active transceivers");
+
+        // Actually set it
+        nttManagerChain1.setThreshold(1);
+        nttManagerChain2.setThreshold(1);
+        nttManagerChain3.setThreshold(1);
+
+        // On chain 1, set the threshold to chain 3 to be different.
+        nttManagerChain1.setPerChainThreshold(chainId3, 2);
+        nttManagerChain3.setPerChainThreshold(chainId1, 2);
```

Comment:
Does this mean that the per-chain threshold should be set to the same value in both directions? Or is it just a coincidence that the threshold is set to 2 in both c1 -> c3 and c3 -> c1?

Reply:
Please take another look since this has all changed.